### PR TITLE
Isak/finalize

### PR DESCRIPTION
**Pull request by IsakFalk**
Cleanup branch

### CODE REVIEW

**Comment by IsakFalk** [comment]
Will merge once finished.

File: Makefile [hunk 1]
```diff
@@ -1,3 +1,5 @@
+GENNI_VERSION=0.1.2
+
 # Format all files using autoflake / isort / black
 remove_unused_imports:
 	autoflake genni --recursive --in-place --remove-unused-variables --remove-all-unused-imports --ignore-init-module-imports
```
Comment (jinxmirror13) [inline Makefile:5]: not on requirements list/inherent in the mac env...

File: Makefile [hunk 2]
```diff
@@ -20,3 +22,7 @@ upload_package: format_package build_package
 # Conda env
 create_conda_env_file:
 	 conda env export > environment.yml
```
Comment (jinxmirror13) [inline Makefile:24]: can we have a pip option pls?

File: Makefile [hunk 3]
```diff
@@ -20,3 +22,7 @@ upload_package: format_package build_package
...
+
+# Reinstall genni
+reinstall_genni: build_package
+	pip uninstall genni -y && pip install dist/genni-$(GENNI_VERSION)*.whl
```
Comment (jinxmirror13) [inline Makefile:28]: there is no pip install command and this pip did not work for me

File: README.md [hunk 4]
```diff
@@ -4,45 +4,49 @@
...
+## Installing
 
-Our package is designed to run in Python 3.6 and pip version 20.2.4....
+Genni is available from PyPI [here](https://pypi.org/project/genni/). In order
+to install simply use `pip`
 
-```
-pip install -r requirements.txt
+```sh
+pip install genni
```
Comment (jinxmirror13) [inline README.md:15]: this does not work...  work for me, get this error when I run 

```suggestion
python setup.py install
```

Error:

```
...
Adding genni 1.0.0 to easy-install.pth file

Installed ***/local/python3.7/site-packages/genni-1.0.0-py3.7.egg
Processing dependencies for genni==1.0.0
Searching for ray[tune]
Reading https://pypi.org/simple/ray/
Downloading https://files.pythonhosted.org/packages/b9/fb/********/ray-1.0.1.post1-***.whl#sha256=********
Best match: ray 1.0.1.post1
Processing ray-1.0.1.post1***.whl
Installing ray-1.0.1.post1-****.whl to ***/local/python3.7/site-packages
Adding ray 1.0.1.post1 to easy-install.pth file

Installed ***/local/python3.7/site-packages/***********************.egg
Traceback (most recent call last):
  File "***/local/python3.7/site-packages/pkg_resources/__init__.py", line 2748, in requires
    deps.extend(dm[safe_extra(ext)])
KeyError: 'tune'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "setup.py", line 29, in <module>
    "tqdm",
  File "***/local/python3.7/site-packages/setuptools/__init__.py", line 144, in setup
    return distutils.core.setup(**attrs)
  File "***/local/python/3.7.3/Frameworks/Python.framework/Versions/3.7/lib/python3.7/distutils/core.py", line 148, in setup
    dist.run_commands()
  File "***/local/python/3.7.3/***/distutils/dist.py", line 966, in run_commands
    self.run_command(cmd)
  File "***/local/python/3.7.3/***/distutils/dist.py", line 985, in run_command
    cmd_obj.run()
  File "***/local/python3.7/site-packages/setuptools/command/install.py", line 67, in run
    self.do_egg_install()
  File "***/local/python3.7/site-packages/setuptools/command/install.py", line 117, in do_egg_install
    cmd.run(show_deprecation=False)
  File "***/local/python3.7/site-packages/setuptools/command/easy_install.py", line 425, in run
    self.easy_install(spec, not self.no_deps)
  File "***/local/python3.7/site-packages/setuptools/command/easy_install.py", line 674, in easy_install
    return self.install_item(None, spec, tmpdir, deps, True)
  File "***/local/python3.7/site-packages/setuptools/command/easy_install.py", line 721, in install_item
    self.process_distribution(spec, dist, deps)
  File "***/local/python3.7/site-packages/setuptools/command/easy_install.py", line 766, in process_distribution
    [requirement], self.local_index, self.easy_install
  File "***/local/python3.7/site-packages/pkg_resources/__init__.py", line 795, in resolve
    new_requirements = dist.requires(req.extras)[::-1]
  File "***/local/python3.7/site-packages/pkg_resources/__init__.py", line 2751, in requires
    "%s has no such extra feature %r" % (self, ext)
pkg_resources.UnknownExtra: ray 1.0.1.post1 has no such extra feature 'tune'
```

File: genni.yml [hunk 5]
```diff
@@ -1 +1 @@
-genni_home: "/home/isak/life/references/projects/phd/do-not-circulate/GENNI_public/"
```
Comment (jinxmirror13) [inline genni.yml:1]: Optional: remove this commit from all history of the repo for privacy and add `genni.yml` to `.gitignore`

File: genni/griding/get_grid.py [hunk 6]
```diff
@@ -61,12 +63,18 @@ def main(experiment_folder, exp_id, config, data_loader):
...
+    cache_folder = cache_data(
         experiment_folder,
         "{}d_loss".format(len(basis_vectors)),
         grid_vals,
         meta_dict=config,
         time_stamp=True,
     )
 
+    config["cache_folder"] = cache_folder
+    config["cache_name"] = {
```
Comment (jinxmirror13) [inline genni/griding/get_grid.py:75]: noice

**Review by jinxmirror13 — CHANGES_REQUESTED** [review]
Could not install, left comments on a few things... Love the improved readme and clean notebook! 

FYSA some of your file paths are shown in the notebooks and in the git history... left some comments on them in case you care about that being public. Can suppress warnings with this:

https://stackoverflow.com/questions/9031783/hide-all-warnings-in-ipython

File: Makefile (hunk 2 above)
Comment (jinxmirror13) [reply Makefile:24]: rather, no pip option in makefile, but not sure that matters since it is in the `README.md`

File: Makefile (hunk 1 above)
Comment (IsakFalk) [reply Makefile:5]: Not sure what you mean by this. This is to reinstall it easily while testing. To be honest could remove this.

File: Makefile (hunk 2 above)
Comment (IsakFalk) [reply Makefile:24]: I have a pip option (but it doesn't work apparently). The problem is ray which has caused a headache from the start. Might look at this a bit more tonight.

The Makefile is only for building and deploying. It's not for users.

File: Makefile (hunk 3 above)
Comment (IsakFalk) [reply Makefile:28]: Not for users.

Comment (IsakFalk) [reply Makefile:28]: For this to work you would have to build the package. I use poetry for that.

File: README.md (hunk 4 above)
Comment (IsakFalk) [reply README.md:15]: We don't want to do `python setup.py install` unless we can avoid it. I'll try to fix the error.

Comment (IsakFalk) [reply README.md:15]: It should work now, I've made the package have all of the dependencies needed to run by just doing pip install.

Note that I've used python=3.8 so we might want to enforce that. I know if fails for python=3.9

File: genni.yml (hunk 5 above)
Comment (IsakFalk) [reply genni.yml:1]: Nothing private, I don't mind just keeping it.

File: Makefile (hunk 1 above)
Comment (IsakFalk) [reply Makefile:5]: For the warnings, I can fix them by using the correct yaml load (`safe_load` vs `load`). Will update at some point.

**Comment by jinxmirror13** [comment]
Progress! Fixed my python environment and also using `poetry`!! (LOVE IT) but now running into other issues 😬  Looks like i get a `ValueError` from `ray/tune/tune.py`...

```
...
(pid=33543) ValueError: Invalid return or yield value. Either return/yield a single number or a dictionary object in your trainable function.
...
Number of trials: 1/1 (1 ERROR)
+--------------------+----------+-------+--------------+--------------+
| Trial name         | status   | loc   |   batch_size |   model_seed |
|--------------------+----------+-------+--------------+--------------|
| lambda_c1392_00000 | ERROR    |       |          256 |           10 |
+--------------------+----------+-------+--------------+--------------+
Number of errored trials: 1
+--------------------+--------------+-------------------------------------------------------------------------------------------------------------------------------------------+
| Trial name         |   # failures | error file                                                                                                                                |
|--------------------+--------------+-------------------------------------------------------------------------------------------------------------------------------------------|
| lambda_c1392_00000 |            1 | /***/ray_results/lambda_2020-12-02_16-59-35/lambda_c1392_00000_0_batch_size=256,model_seed=10_2020-12-02_16-59-35/error.txt |
+--------------------+--------------+-------------------------------------------------------------------------------------------------------------------------------------------+

Traceback (most recent call last):
  File "scripts/experiment.py", line 90, in <module>
    tune.run(
  File "/***/GENNI_public/.venv/lib/python3.8/site-packages/ray/tune/tune.py", line 434, in run
    raise TuneError("Trials did not complete", incomplete_trials)
ray.tune.error.TuneError: ('Trials did not complete', [lambda_c1392_00000])
```

**Pull request merged**